### PR TITLE
Consumer arg support for shovels

### PR DESCRIPTION
**Pull request by viktorerlingsson**
### WHAT is this pull request doing?
Shovels handle `src-consumer-args` to enable them to handle stream queues. 

### HOW can this pull request be tested?
Create a shovel for a stream queue via API with x-stream-offset

### CODE REVIEW

File: src/lavinmq/shovel/shovel.cr [hunk 1]
```diff
@@ -41,7 +41,8 @@ module LavinMQ
 
       def initialize(@name : String, @uri : URI, @queue : String?, @exchange : String? = nil,
                      @exchange_key : String? = nil,
-                     @delete_after = DEFAULT_DELETE_AFTER, @prefetch = DEFAULT_PREFETCH, @ack_mode = DEFAULT_ACK_MODE,
+                     @delete_after = DEFAULT_DELETE_AFTER, @prefetch = DEFAULT_PREFETCH,
+                     @ack_mode = DEFAULT_ACK_MODE, @consumer_args = {} of String => JSON::Any || nil,
```
Comment (spuun) [inline src/lavinmq/shovel/shovel.cr:45]: Any reason to have `@consumer_args` as a instance variable? Looks like it's only used to populate `@args`?

And the `|| nil` part seems odd. Will it ever do anything? Default value is empty hash. Or do you want `@consumer_args : Hash(String, JSON::Any)? = nil`?

File: src/lavinmq/shovel/shovel.cr [hunk 2]
```diff
@@ -60,6 +61,10 @@ module LavinMQ
         if @queue.nil? && @exchange.nil?
           raise ArgumentError.new("Shovel source requires a queue or an exchange")
         end
+        @args = AMQ::Protocol::Table.new
+        @consumer_args.try &.each do |k, v|
+          @args[k] = v
```
Comment (spuun) [inline src/lavinmq/shovel/shovel.cr:66]: How will `JSON::Any` be handled from here? Should it be `v.as_s?` or something?

File: src/lavinmq/shovel/shovel.cr (hunk 1 above)
Comment (viktorerlingsson) [reply src/lavinmq/shovel/shovel.cr:45]: Yeah, looks much better that way 👍

File: src/lavinmq/shovel/shovel.cr (hunk 2 above)
Comment (viktorerlingsson) [reply src/lavinmq/shovel/shovel.cr:66]: I don't `v.as_s?` is needed for it to work, but it's probably best to be specific

**Comment by viktorerlingsson** [comment]
Moved the definitions upload stuff to a separate PR to keep this PR more focused

File: src/lavinmq/shovel/shovel.cr (hunk 2 above)
Comment (spuun) [reply src/lavinmq/shovel/shovel.cr]: Oh, `AMQ::Protocol::Table` accepts `JSON::Any` as value. Didn't know that (and I guess it compiled for you 🤦).

**Review by carlhoerberg — APPROVED** [review]

**Pull request merged**